src/pinned_memory_manager.h

```diff
@@ -99,6 +108,8 @@ class PinnedMemoryManager {
 
   static std::unique_ptr<PinnedMemoryManager> instance_;
   static uint64_t pinned_memory_byte_size_;
+  static std::vector<std::shared_ptr<PinnedMemory>>
```

Comment:
You should add a static mutex to protect the allocated_pinned_memory_buffers_  vector.

Reply:
Done 👍